src/Service/Platform.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Form\Service;
+
+use DateTime;
+use stdClass;
+
+class Platform
+{
+    public function datetime(): DateTime
+    {
+        return new DateTime();
+    }
+
+    /**
+     * @param array<mixed>|object $array
+     * @return array<string,mixed>|array<mixed> $array
+     */
+    public function objectToArrayRecursive(mixed $array): array
+    {
+        if (is_array($array)) {
+            foreach ($array as $key => $value) {
+                if (is_array($value)) {
+                    $array[$key] = $this->objectToArrayRecursive($value);
+                }
+                if ($value instanceof stdClass) {
+                    $array[$key] = $this->objectToArrayRecursive((array) $value);
+                }
+            }
+        }
+        if ($array instanceof stdClass) {
+            return $this->objectToArrayRecursive((array) $array);
+        }
+        /** @var array<string,mixed>|array<mixed> $array */
+        return $array;
+    }
+
+    /**
+     * @param array<mixed>|array<string,mixed> $array
+     * @throws \JsonException
+     */
+    public function arrayToObjectRecursive(array $array): object
+    {
+        $json = json_encode($array, JSON_THROW_ON_ERROR);
+        return (object) json_decode($json, false, 512, JSON_THROW_ON_ERROR);
+    }
```

Comment:
These don't seem like `Platform` reliant functions; It looks a bit like this is beeing (ab)used as a `Util` class.  
Are these functions really that commonly required? if so - why?

Reply:
The methods are only used once. I have pulled them out so that I can test them individually.
I don't want an `Util` class. What could I call them?